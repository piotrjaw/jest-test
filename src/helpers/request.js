import axios from 'axios';

const contructUrl = (url) => `http://localhost:5000${url}`;

export const del = (url) => axios.delete(contructUrl(url));

export const get = (url) => axios.get(contructUrl(url));

export const post = (url, data) => axios.post(contructUrl(url), data);

export const put = (url, data) => axios.put(contructUrl(url), data);
