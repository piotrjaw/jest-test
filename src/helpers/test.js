import React from 'react';

export const getWrapper = ({ Component, method, props }) => {
  return method(
    <Component
      { ...props }
    >
      { props.children }
    </Component>
  );
};

export const getMockStore = (mockFn, state = {}) => ({
  subscribe: mockFn(),
  dispatch: mockFn(),
  getState: mockFn(() => ({
    ...state,
  })),
});
