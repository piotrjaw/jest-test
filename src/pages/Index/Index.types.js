export const CREATE_TODO_PENDING = 'Index/CREATE_TODO_PENDING';
export const CREATE_TODO_SUCCESS = 'Index/CREATE_TODO_SUCCESS';
export const DELETE_TODO_PENDING = 'Index/DELETE_TODO_PENDING';
export const DELETE_TODO_SUCCESS = 'Index/DELETE_TODO_SUCCESS';
export const GET_TODOS_PENDING = 'Index/GET_TODOS_PENDING';
export const GET_TODOS_SUCCESS = 'Index/GET_TODOS_SUCCESS';
export const UPDATE_TODO_PENDING = 'Index/UPDATE_TODO_PENDING';
export const UPDATE_TODO_SUCCESS = 'Index/UPDATE_TODO_SUCCESS';
