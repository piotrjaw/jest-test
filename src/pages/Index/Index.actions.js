import * as TYPES from './Index.types';
import { del, get, put, post } from '../../helpers/request';

const createTodoPending = () => ({
  type: TYPES.CREATE_TODO_PENDING
});

const createTodoSuccess = () => ({
  type: TYPES.CREATE_TODO_SUCCESS
});

const deleteTodoPending = () => ({
  type: TYPES.DELETE_TODO_PENDING
});

const deleteTodoSuccess = () => ({
  type: TYPES.DELETE_TODO_SUCCESS
});

const getTodosPending = () => ({
  type: TYPES.GET_TODOS_PENDING
});

export const getTodosSuccess = (data) => ({
  type: TYPES.GET_TODOS_SUCCESS,
  data
});

const updateTodoPending = () => ({
  type: TYPES.UPDATE_TODO_PENDING
});

const updateTodoSuccess = () => ({
  type: TYPES.UPDATE_TODO_SUCCESS
});

export const createTodo = (createObj) => (dispatch) => {
  if (!createObj.text) return Promise.reject();
  dispatch(createTodoPending());
  return post('/todos/', createObj)
    .then((res) => dispatch(createTodoSuccess()));
}

export const deleteTodo = (deleteObj) => (dispatch) => {
  dispatch(deleteTodoPending());
  return del('/todos/', deleteObj)
    .then((res) => dispatch(deleteTodoSuccess()));
};

export const getTodos = () => (dispatch) => {
  dispatch(getTodosPending());
  return get('/todos/')
    .then(({ data }) => dispatch(getTodosSuccess(data)));
};

export const updateTodo = (updateObj) => (dispatch) => {
  dispatch(updateTodoPending());
  return put('/todos/', updateObj)
    .then((res) => dispatch(updateTodoSuccess()));
};
