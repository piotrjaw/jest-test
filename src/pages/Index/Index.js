import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as IndexActions from './Index.actions';
import List from '../../components/List';

class IndexComponent extends PureComponent {
  render() {
    return (
      <div>Index</div>
    );
  }
}

const mapStateToProps = ({ index }) => ({

});

const mapDispatchToProps = (dispatch) => ({
  actions: bindActionCreators(IndexActions, dispatch)
});

const Index = connect(
  mapStateToProps,
  mapDispatchToProps
)(IndexComponent);

export default Index;
