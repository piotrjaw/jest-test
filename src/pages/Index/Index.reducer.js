import * as TYPES from './Index.types';

const initialState = { todos: [] };

export default (state = initialState, action) => {
  switch (action.type) {
    case TYPES.GET_TODOS_SUCCESS:
      return {
        todos: action.data
      }
    default:
      return {
        todos: state.todos || []
      };
  }
};
