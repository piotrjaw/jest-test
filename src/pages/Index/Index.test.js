import { mount, shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { getMockStore, getWrapper } from '../../helpers/test';
import Index from './Index';

const setupWrapper = (config = {}) => {
  const { method = shallow, props = {} } = config;
  return getWrapper({
    Component: Index,
    method,
    props: {
      store: getMockStore(jest.fn, {
        index: { todos: [] }
      }),
      ...props
    }
  });
};

let wrapper;

jest
  .mock('./Index.actions', () => ({
    getTodos: jest.fn()
  }));

beforeEach(() => {
  wrapper = setupWrapper();
});

afterEach(() => {
  expect(toJSON(wrapper.render())).toMatchSnapshot();
  wrapper = null;
  jest.resetAllMocks();
});

describe('Index page', () => {
  it('should render', () => {
    expect(wrapper.exists()).toBe(true)
  });

  it('should call getTodos', () => {
    wrapper = setupWrapper({ method: mount });
    expect(require('./Index.actions').getTodos).toHaveBeenCalledTimes(1);
  });
});
