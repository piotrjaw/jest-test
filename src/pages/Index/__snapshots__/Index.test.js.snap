// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page should call getTodos 1`] = `
<div>
  <div>
    <input
      type="text"
      value=""
    />
    <button
      type="button"
    >
      Add task
    </button>
  </div>
  <div
    class="jt-List"
  />
</div>
`;

exports[`Index page should render 1`] = `
<div>
  <div>
    <input
      type="text"
      value=""
    />
    <button
      type="button"
    >
      Add task
    </button>
  </div>
  <div
    class="jt-List"
  />
</div>
`;
